### PR TITLE
[NTH-59] - Signup page

### PR DESCRIPTION
**Pull request by ahmedmurtaja**
- make a signup page for patients & therapist
- handle rendering additional fields
- handle scroll behavior 
- fix styling problem and image resizing

### CODE REVIEW

**Review by raghdarizq — APPROVED** [review]

**Review by YasserAmro0 — APPROVED** [review]

File: src/pages/signup/index.tsx [hunk 1]
```diff
@@ -1,5 +1,225 @@
...
+} from './classes';
+import './style.css';
+
+const handleFileUpload = (_event: React.ChangeEvent<HTMLInputElement>) => {
+  // eslint-disable-next-line no-console
+  console.log('File uploaded successfully.');
+};
+
+const Signup = () => {
+  const [userType, setUserType] = useState('');
```
Comment (LinaYahya) [inline src/pages/signup/index.tsx]: let us have a default value as a user

File: src/pages/signup/index.tsx [hunk 2]
```diff
@@ -1,5 +1,225 @@
...
+  const [showAdditionalFields, setShowAdditionalFields] = useState(false);
```
Comment (LinaYahya) [inline src/pages/signup/index.tsx]: No need for this as a state, actually we can render additional inputs based on user type

File: src/pages/signup/index.tsx [hunk 3]
```diff
@@ -1,5 +1,225 @@
...
+                value="therapist"
+                control={<Radio />}
+                label="therapist"
+                labelPlacement="end"
+              />
+              <FormControlLabel
+                style={{ margin: '10px' }}
+                value="patient"
+                control={<Radio />}
+                label="patient"
```
Comment (LinaYahya) [inline src/pages/signup/index.tsx]:
```suggestion
                label="user"
```
And I'd prefer to have user radio btn before therapist

File: src/pages/signup/index.tsx [hunk 4]
```diff
@@ -1,5 +1,225 @@
...
+                labelPlacement="end"
+              />
+            </RadioGroup>
+            <TextField
+              margin="normal"
+              required
+              sx={textFieldStyle}
+              id="username"
+              name="username"
+              label="username"
```
Comment (LinaYahya) [inline src/pages/signup/index.tsx]:
```suggestion
              label="Full Name"
```

File: src/pages/signup/index.tsx [hunk 5]
```diff
@@ -1,5 +1,225 @@
...
+              helperText={formik.touched.password && formik.errors.password}
+            />
+            <TextField
+              margin="normal"
+              required
+              autoFocus
+              sx={textFieldStyle}
+              id="confirmPassword"
+              name="confirmPassword"
+              label="confirmPassword"
```
Comment (LinaYahya) [inline src/pages/signup/index.tsx]:
```suggestion
              label="Confirm Password"
```

File: src/pages/signup/schema.ts [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  confirmPassword: yup
+    .string()
+    .required('Confirm Password is required')
+    .oneOf([yup.ref('password')], 'Passwords must match'),
+  major: yup
+    .string()
+    .required('Major is required'),
+  hourlyRate: yup
+    .number()
+    .required('Hourly Rate is required'),
```
Comment (LinaYahya) [inline src/pages/signup/schema.ts:30]: can we have a check for CV and profile image, OR Just to handle it later after take decision of how to upload it?
@ahmedmurtaja

File: src/pages/signup/index.tsx [hunk 7]
```diff
@@ -1,5 +1,225 @@
...
+        <>
+          <TextField
+            margin="normal"
+            required
+            autoFocus
+            sx={textFieldStyle}
+            id="major"
+            name="major"
+            label="major"
+            type="major"
```
Comment (LinaYahya) [inline src/pages/signup/index.tsx]:
```suggestion
            type="major"
```
🤔

File: src/pages/signup/index.tsx [hunk 8]
```diff
@@ -1,5 +1,225 @@
...
+          />
+          <TextField
+            margin="normal"
+            required
+            autoFocus
+            sx={textFieldStyle}
+            id="hourlyRate"
+            name="hourlyRate"
+            label="hourlyRate"
+            type="hourlyRate"
```
Comment (LinaYahya) [inline src/pages/signup/index.tsx]: input type is not associated with it's name, Such common input types (email, name, number, password)

File: src/pages/signup/index.tsx [hunk 9]
```diff
@@ -1,5 +1,225 @@
...
+              label="Password"
+              type="password"
+              onChange={formik.handleChange}
+              error={formik.touched.password && Boolean(formik.errors.password)}
+              helperText={formik.touched.password && formik.errors.password}
+            />
+            <TextField
+              margin="normal"
+              required
+              autoFocus
```
Comment (LinaYahya) [inline src/pages/signup/index.tsx]:
```suggestion
```

File: src/pages/signup/index.tsx [hunk 10]
```diff
@@ -1,5 +1,225 @@
...
+              helperText={formik.touched.username && formik.errors.username}
+            />
+            <TextField
+              margin="normal"
+              required
+              sx={textFieldStyle}
+              id="email"
+              name="email"
+              label="Email"
+              autoFocus
```
Comment (LinaYahya) [inline src/pages/signup/index.tsx]:
```suggestion
```
AutoFoxus gonna led to focus last input which I find it a bit annoying, SO I'd prefer to delete this attribute for all inputs

**Review by LinaYahya — CHANGES_REQUESTED** [review]
Good Job Ahmed, A small notes to take in consideration, 
As well as we need to have a show/hide password functionality 
[Input Adornments](https://mui.com/material-ui/react-text-field/#input-adornments), and I guess changes u made affect login background image, can u check?

File: src/pages/signup/index.tsx (hunk 1 above)
Comment (ahmedmurtaja) [reply src/pages/signup/index.tsx]: done

File: src/pages/signup/index.tsx (hunk 2 above)
Comment (ahmedmurtaja) [reply src/pages/signup/index.tsx]: done

File: src/pages/signup/index.tsx (hunk 3 above)
Comment (ahmedmurtaja) [reply src/pages/signup/index.tsx]: done

File: src/pages/signup/index.tsx (hunk 4 above)
Comment (ahmedmurtaja) [reply src/pages/signup/index.tsx]: done

File: src/pages/signup/index.tsx (hunk 5 above)
Comment (ahmedmurtaja) [reply src/pages/signup/index.tsx]: done

File: src/pages/signup/schema.ts (hunk 6 above)
Comment (ahmedmurtaja) [reply src/pages/signup/schema.ts:30]: I think we can handle it later in the S3 task, but for now, I think of using [React-Toastify](https://fkhadra.github.io/react-toastify/) for handling successful or failed file upload

**Comment by ahmedmurtaja** [comment]
> Good Job Ahmed, A small note to take into consideration, As well as we need to have a show/hide password functionality [Input Adornments](https://mui.com/material-ui/react-text-field/#input-adornments), and I guess the changes u made affect login background image, can u check?

Sorry, I changed the login style by mistake (I fixed it now).

I will make a task to fix the password field on Login Page also after fixing it in this branch

**Review by MohannadSabaa — APPROVED** [review]

**Review by LinaYahya — APPROVED** [review]

**Pull request merged**